File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
  const plus = function (sum, numbers, i) {
    sum += numbers[i];
    if (i == numbers.length - 1 && sum == target) {
      cnt += 1;
    } else if (i != numbers.length - 1) {
      plus(sum, numbers, i + 1);
      minus(sum, numbers, i + 1);
    }
  };
  const minus = function (sum, numbers, i) {
    sum -= numbers[i];
    if (i == numbers.length - 1 && sum == target) {
      cnt += 1;
    } else if (i != numbers.length - 1) {
      plus(sum, numbers, i + 1);
      minus(sum, numbers, i + 1);
    }
  };
  let cnt = 0;
  plus(0, numbers, 0);
  minus(0, numbers, 0);

  return cnt;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    
    def dfs(sums, n):
        if n == len(numbers):
            if sums == target:
                return 1
            else:
                return 0
        plus = dfs(sums + numbers[n], n+1)
        minus = dfs(sums - numbers[n], n+1)
        
        return plus + minus

    return dfs(0, 0)